libs/cluster/Session/ClusterSession.cs

```diff
@@ -171,20 +161,23 @@ public void SetUser(User user)
         }
 
         /// <summary>
-        /// Performs @admin command group permission checks for the current user and the given command.
-        /// (NOTE: This function is temporary until per-command permissions are implemented)
+        /// Performs permission checks for the current user and the given command.
+        /// (NOTE: This function does not check keyspaces)
         /// </summary>
-        /// <param name="bufSpan">Buffer containing the current command in RESP3 style.</param>
+        /// <param name="cmd">Command be processed</param>
+        /// <param name="subCommand">Id (w.r.t. <see cref="RespCommandsInfo"/>) of any sub command - will be 0 if no sub command is being processed</param>
         /// <param name="count">Number of parameters left in the command specification.</param>
         /// <param name="processingCompleted">Indicates whether the command was completely processed, regardless of whether execution was successful or not.</param>
         /// <returns>True if the command execution is allowed to continue, otherwise false.</returns>
-        bool CheckACLAdminPermissions(ReadOnlySpan<byte> bufSpan, int count, out bool processingCompleted)
+        bool CheckACLPermissions(RespCommand cmd, byte subCommand, int count, out bool processingCompleted)
```

Comment:
I had taken this route because such a buffer wasn't available everywhere I needed to slap an ACL check.  Now that I've DRY'd that up a bit, I agree taking a buffer is cleaner - I'll take a whack at it.